app/scripts/modules/core/instance/instanceType.service.ts

```diff
@@ -85,7 +86,7 @@ export class InstanceTypeService {
 
   private getInstanceTypeCategory(cloudProvider: string, instanceType: string): ng.IPromise<IInstanceTypeCategory> {
     return this.getCategories(cloudProvider).then((categories: IInstanceTypeCategory[]) => {
-      return (categories || []).find(c => c.families.some(f => f.instanceTypes.some(i => i.name === instanceType)));
+      return (categories || []).find(c => c.families.some(f => f.instanceTypes.some(i => i.name === instanceType || (i.nameRegex && i.nameRegex.test(instanceType)))));
```

Comment:
I see what you're doing here, but this strikes me as funky.  We're adding a conditional that only looks at a property set by one cloud provider.

In OOP land, a class would override a method to provide the necessary implementation/business logic, but we're not _really_ in OOP land and we'd have to create a whole InstanceTypeServiceProvider mechanism to have the code serve up the "right" service...

I wonder if we could take advantage of the extracted settings by @jrsquared and customize this there instead?  Or is that totally overkill?